### DISCUSSION TITLE
Ignoring unhandled files matched by pattern (alternative to Prettier’s `--ignore-unknown`)

### DISCUSSION THREAD

**Discussion by valtlai** [rome/tools#4267] [Suggestions]
With Prettier, I can use the following to format staged files:

```json
"lint-staged": {
  "*": "prettier --write --ignore-unknown"
},
```

Here, the option [--ignore-unknown](https://prettier.io/docs/en/cli.html#--ignore-unknown) makes unknown file types to be silently ignored. This way I don’t have to list all the supported file extensions in the glob pattern of `lint-staged`, like `*{css,html,js,json,md,ts}`.

Rome seems to have no such option. Could it be added? Note that the potential option should apply to `rome check` in addition to `rome format`.

**Comment by Th3S4mur41** [comment]
+1 on this one...

When any of the stage file is of a format that is currently not supported by rome, this ends up with in **× unhandled file type** error

While the workaround to list all supported file type is pretty simple to apply, it has one major downside: It would need to be updated manually every time rome adds support for new file type.

An option like [--ignore-unknown in prettier](https://prettier.io/docs/en/cli.html#--ignore-unknown) would definitely be a nicer touch

**Reply by obennett-m** [reply]
It is also a problem when using rome to format files being committed to git, and it gets passed a (husky or otherwise) shell script. Since pre-commit hooks have no extension the unhandled files error shows up. Very annoying when you (are trying to) rebase a long list of commit.

**Comment by ematipico** [comment]
That sounds like a reasonable request, and I suppose ALL commands - `rome format`, `rome check`, `rome ci` - should benefit from this new argument.
